web/src/redux/slices/TestDefinition.slice.ts

```diff
@@ -74,6 +76,12 @@ const testDefinitionSlice = createSlice({
     setAssertionResults(state, {payload}: PayloadAction<TAssertionResults>) {
       state.assertionResults = payload;
     },
+    clearAffectedSpans(state) {
```

Comment:
Maybe a change for the future but, I was thinking of decoupling both the affected spans and the selected span into a separated slice. As you can see, the setSelectedSpan function is being drilled into multiple component levels. We can solve it by having a central place for both.

Reply:
Yeah, like the idea. I can work on that after the MVP priorities.